### PR TITLE
Use relative path for CSS

### PR DESCRIPTION
**Pull request by ghost**
We should use a relative path for the CSS URL for sites that do not run from the root path (eg. https://example.com/mywebsite)

### CODE REVIEW

**Comment by rluetzner** [comment]
Does this work correctly, though? I've had trouble with sites hosted on a subpath, when I navigate to another subpath, i.e. `https://example.com/website/posts`.

If that works correctly, I have no objections. 🙂

**Comment by ghost** [comment]
Yes,  https://wheresalice.github.io/hugo-webring/members/ is using my fork of the theme from https://github.com/wheresalice/hugo-webring

**Comment by rluetzner** [comment]
Sorry for the scepticism. 😅 It's very possible that I was simply too stupid when I tried it last.

----

Ah. You're using the RelPath shortcode. That's where I messed up.

Can you update your fork? I just noticed in the diff, that merging this would downgrade the readable-css version. 🙂

**Comment by ghost** [comment]
Sorry, I don't know how I managed to bring in the old version

I've updated my PR now

**Comment by rluetzner** [comment]
Thanks. 🙂

Sorry for bringing up one more thing, but this will also break the GitHub workflow that automatically updates readable-css. I didn't expect that this line would change so soon (or at all). At one point in the script '.github/scripts/update-readable-css.sh', we use `sed` to replace the version in the line. The current implementation stupidly replaces the rest of the line as well, ending with `/>` which is now no longer correct.

I'd appreciate it if you had a quick look, although I can also prepare a follow up PR for when this is merged. We just shouldn't forget it, otherwise this will break as soon as readable.css is updated.

**Comment by benjaminbhollon** [comment]
> this will break as soon as readable.css is updated

*updates readable.css and begins maniacal laughter*

**Comment by ghost** [comment]
good catch, I've updated the PR to use `$CURRENT_VERSION` when making the replacement

File: .github/scripts/update-readable-css.sh [hunk 1]
```diff
@@ -35,7 +35,7 @@ else
     curl -s "https://codeberg.org/Freedom-to-Write/readable.css/raw/tag/${TAG_VERSION}/readable.css" > static/css/readable.css
     curl -s "https://codeberg.org/Freedom-to-Write/readable.css/raw/tag/${TAG_VERSION}/readable.min.css" > static/css/readable.min.css
     VERSION=$(echo "${TAG_VERSION}" | sed 's/v//')
-    sed -i "s/readable.min.css?v=.*/readable.min.css?v=${VERSION}\">/" layouts/partials/head.html
+    sed -i "s/readable.min.css?v=${CURRENT_VERSION}/readable.min.css?v=${VERSION}/" layouts/partials/head.html
```
Comment (rluetzner) [inline .github/scripts/update-readable-css.sh]: That's a great idea. I feel a bit stupid now. 😅

But be careful, that the CURRENT_VERSION and TAG_VERSION start with a 'v'. That's why we use a different VERSION in this replacement.

An easy way to fix this is something like `CURRENT_VERSION=$(echo $CURRENT_VERSION | sed 's/v//')`.

An alternative would be to never add the 'v' prefix to CURRENT_VERSION in the first place and prefix it explicitly where it's needed.

**Review by rluetzner — COMMENTED** [review]
*Enter Columbo*: just one more thing. 😅

**Comment by ghost** [comment]
why not keep it simple and use the version with the 'v' in everywhere?

Updated PR to do just that - unless there's a reason to strip the 'v'?

**Comment by rluetzner** [comment]
Umm. Honestly, I'm not sure. I always assumed that this specific syntax is needed, which is why I've stripped off the 'v' when I implemented the pipeline. Maybe @benjaminbhollon or @cjtheham can answer this. 😬

**Comment by cjtheham** [comment]
So am I understanding the question right as to basically is the `?v=**v**1.1.0` second v required or not? Basically the actual syntax of everything after the `?` doesn't matter. It's a trick that Ben cleverly mentions in the readable docs that I picked up, basically just make sure that the query changes with every version update. Doing this does nothing to change the actual CSS file itself, but makes the browser do a full new request because it's a new URL, so it won't cache the old version. So v or no v, it would still work. I think I dropped the v just because it made sense in my brain at the time, but it should still work just fine with it added. Up to you guys if you want to drop it or keep it!

**Comment by rluetzner** [comment]
That's a neat trick. I'll look over the code one last time, but I'm pretty sure it's alright. :slightly_smiling_face:

**Review by rluetzner — APPROVED** [review]
Yup. Awesome.

Thank you. :slightly_smiling_face:

**Comment by cjtheham** [comment]
Thanks for the contribution @wheresalice - it's been merged! Hope this helps development on your webring site.

**Comment by benjaminbhollon** [comment]
> Doing this does nothing to change the actual CSS file itself, but makes the browser do a full new request because it's a new URL, so it won't cache the old version

Exactly right. It doesn't really matter *what* it is so long as it changes for each release to force a cache refresh. An absolutely solid way would perhaps to use a hash of the content of the CSS file, to make sure cache refreshes whenever the file changes, but I think that's overkill for this application. ;)

**Pull request merged**